pkg/hubble/exporter/aggregator_test.go

```diff
@@ -0,0 +1,414 @@
+		aggregator := NewAggregator(hivetest.Logger(t))
+		aggregator.Add(&v1.Event{Event: nil})
+		assert.Empty(t, aggregator.m)
+	})
+
+	t.Run("non-flow event", func(t *testing.T) {
+		aggregator := NewAggregator(hivetest.Logger(t))
+		aggregator.Add(&v1.Event{Event: &observerpb.AgentEvent{}})
+		assert.Empty(t, aggregator.m)
+	})
+
+	t.Run("unknown direction flows", func(t *testing.T) {
+		fieldMask, err := fieldmaskpb.New(&flowpb.Flow{}, "verdict")
+		require.NoError(t, err)
+
+		fieldAgg, err := fieldaggregate.New(fieldMask)
+		require.NoError(t, err)
+
+		aggregator := NewAggregatorWithFields(fieldAgg, hivetest.Logger(t))
+
+		// Add multiple unknown direction flows
+		aggregator.Add(&v1.Event{
+			Event: &flowpb.Flow{
+				Verdict:          flowpb.Verdict_FORWARDED,
+				TrafficDirection: flowpb.TrafficDirection_TRAFFIC_DIRECTION_UNKNOWN,
+			},
+		})
+		aggregator.Add(&v1.Event{
+			Event: &flowpb.Flow{
+				Verdict:          flowpb.Verdict_FORWARDED,
+				TrafficDirection: flowpb.TrafficDirection_TRAFFIC_DIRECTION_UNKNOWN,
+			},
+		})
+
+		// Verify aggregation
+		assert.Len(t, aggregator.m, 1)
+
+		// Create the expected key for verdict=FORWARDED
+		expectedFlow := &flowpb.Flow{Verdict: flowpb.Verdict_FORWARDED}
+		expectedKey := aggregator.generateAggregationKey(expectedFlow)
+
+		value, exists := aggregator.m[expectedKey]
+		require.True(t, exists, "Expected aggregation key should exist")
+		assert.Equal(t, 0, value.IngressFlowCount)
+		assert.Equal(t, 0, value.EgressFlowCount)
+		assert.Equal(t, 2, value.UnknownDirectionFlowCount)
+	})
+}
+
+func TestAggregateComplexScenario(t *testing.T) {
+	fieldMask, err := fieldmaskpb.New(&flowpb.Flow{}, "destination.pod_name", "source.pod_name")
+	require.NoError(t, err)
+
+	fieldAgg, err := fieldaggregate.New(fieldMask)
+	require.NoError(t, err)
+
+	aggregator := NewAggregatorWithFields(fieldAgg, hivetest.Logger(t))
+
+	for _, event := range getEventList() {
+		aggregator.Add(event)
+	}
+
+	assert.Len(t, aggregator.m, 1)
+
+	for _, value := range aggregator.m {
+		assert.Equal(t, 1, value.IngressFlowCount)
+		assert.Equal(t, 2, value.EgressFlowCount)
+		assert.Equal(t, 0, value.UnknownDirectionFlowCount)
+		assert.NotNil(t, value.ProcessedFlow)
+	}
+}
+
+func TestAggregatorRunFunction(t *testing.T) {
+	t.Run("processes events and resets map", func(t *testing.T) {
+		buf := &bytesWriteCloser{bytes.Buffer{}}
+		log := hivetest.Logger(t)
+
+		opts := DefaultOptions
+		opts.newWriterFunc = func() (io.WriteCloser, error) {
+			return buf, nil
+		}
+
+		err := WithAggregationInterval(100 * time.Millisecond)(&opts)
+		require.NoError(t, err)
+		err = WithFieldAggregate([]string{"verdict"})(&opts)
+		require.NoError(t, err)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		exporter, err := newExporter(ctx, log, opts)
+		require.NoError(t, err)
+		defer exporter.Stop()
+
+		exporter.a.Add(getEventList()[0])
+
+		// Wait for Run cycle
+		time.Sleep(200 * time.Millisecond)
```

Comment:
Using sleeps will cause flakes in CI eventually. Please either use the new Go synctest framework, or add some logic to the aggregator to get a signal/callback/event once the interval is reached, or use the `assert.Eventually` family of functions to wait for the expected behavior.